### PR TITLE
Properly compute device for elementwise operations with CPU scalar tensor

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #93073

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by eellison — APPROVED** [review]

**Review by bdhirsh — APPROVED** [review]
nice catch

File: torch/_prims/__init__.py [hunk 1]
```diff
@@ -368,7 +368,7 @@ def _elementwise_meta(
     device = None
     number = None
     for arg in args_:
-        if isinstance(arg, TensorLike):
+        if isinstance(arg, TensorLike) and not utils.is_cpu_scalar_tensor(arg):
```
Comment (ngimel) [inline torch/_prims/__init__.py:371]: test failures are real (for 0d cpu + 0d cpu I think)

**Pull request closed without merging**